compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -3482,3 +3482,86 @@ fn make_thin_self_ptr<'tcx>(
         ..tcx.layout_of(ty::ParamEnv::reveal_all().and(unit_ptr_ty)).unwrap()
     }
 }
+
+/// Determines if this type permits "raw" initialization by just transmuting some
+/// uninitialized memory into an instance of `T`.
+///
+/// This code is intentionally conservative, and will not detect
+/// * making uninitialized types who have a full valid range (ints, floats, raw pointers)
+/// * uninit invalid &[T] where T has align 1 (only inside arrays)
```

Comment:
```
error: unresolved link to `T`
--> compiler/rustc_middle/src/ty/layout.rs:3491:24
|
3491|/// * uninit invalid &[T] where T has align 1 (only inside arrays)
| ^no item named `T` in scope
|
= note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
= help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
```

I guess you need to put these into backticks ^^

Reply:
Right. Yeah, that's the second time I messed up a docs thing, I really gotta get into the habit of checking x.py doc :)

Will fix now, sorry about that :)

(How expensive would it be for PR CI to run docs? I don't know if I'm just the only one hitting this or if it would be worth adding it to PR CI.)

Reply:
```
workspace: /home/jess/src/rust/Cargo.toml
   Compiling serde v1.0.125
   Compiling rand v0.8.5
   Compiling tempfile v3.2.0
   Compiling serde_json v1.0.59
   Compiling lint-docs v0.1.0 (/home/jess/src/rust/src/tools/lint-docs)
    Finished release [optimized] target(s) in 6.08s
warning: the code example in lint `unfulfilled_lint_expectations` in /home/jess/src/rust/compiler/rustc_lint_defs/src/builtin.rs failed to generate the expected output: did not find lint `unfulfilled_lint_expectations` in output of example, got:

error[E0554]: `#![feature]` may not be used on the beta release channel
 --> lint_example.rs:1:1
  |
1 | #![feature(lint_reasons)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^


error: aborting due to previous error


For more information about this error, try `rustc --explain E0554`.

warning: the code example in lint `unused_allocation` in /home/jess/src/rust/compiler/rustc_lint/src/unused.rs failed to generate the expected output: did not find lint `unused_allocation` in output of example, got:

error[E0554]: `#![feature]` may not be used on the beta release channel
 --> lint_example.rs:1:1
  |
1 | #![feature(box_syntax)]
  | ^^^^^^^^^^^^^^^^^^^^^^^


error: aborting due to previous error


For more information about this error, try `rustc --explain E0554`.

Rustbook (x86_64-unknown-linux-gnu) - rustc
Rustbook (x86_64-unknown-linux-gnu) - cargo
```

known issue? I saw that on master doc runs too. In any case, I couldn't see any failures that seemed to be caused by me, so.... It's probably fine now.

Reply:
> How expensive would it be for PR CI to run docs?

Not too terribly long, but for it to be reliable we can't use stage 0 rustdoc, which means we have to add it to the LLVM-12 builder, which is already kinda slow :/

Reply:
Oh also do you mind opening an issue for that feature gate error? We should fix it, but we *really* should make sure it's tested somewhere and not just ignored.